src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/SpellCheck/LspCSharpSpellCheckRangeProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.LanguageServer.Hosting;
+using Microsoft.AspNetCore.Razor.PooledObjects;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Protocol;
+using Microsoft.CodeAnalysis.Razor.SpellCheck;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.SpellCheck;
+
+internal sealed class LspCSharpSpellCheckRangeProvider(
+    LanguageServerFeatureOptions languageServerFeatureOptions,
+    IClientConnection clientConnection) : ICSharpSpellCheckRangeProvider
+{
+    private readonly LanguageServerFeatureOptions _languageServerFeatureOptions = languageServerFeatureOptions;
+    private readonly IClientConnection _clientConnection = clientConnection;
+
+    public async Task<ImmutableArray<SpellCheckRange>> GetCSharpSpellCheckRangesAsync(DocumentContext documentContext, CancellationToken cancellationToken)
+    {
+        if (!_languageServerFeatureOptions.SingleServerSupport)
+        {
+            return [];
+        }
+
+        var delegatedParams = new DelegatedSpellCheckParams(documentContext.GetTextDocumentIdentifierAndVersion());
+        var delegatedResponse = await _clientConnection.SendRequestAsync<DelegatedSpellCheckParams, VSInternalSpellCheckableRangeReport[]?>(
+            CustomMessageNames.RazorSpellCheckEndpoint,
+            delegatedParams,
+            cancellationToken).ConfigureAwait(false);
+
+        if (delegatedResponse is null)
+        {
+            return [];
+        }
+
+        using var ranges = new PooledArrayBuilder<SpellCheckRange>();
+        foreach (var report in delegatedResponse)
+        {
+            if (report.Ranges is not { } csharpRanges)
+            {
+                continue;
+            }
```

Comment:
Are there expected scenarios when we would continue here?

```suggestion
            if (report.Ranges is not { } csharpRanges)
            {
                continue;
            }
```

Reply:
I wouldn't _expect_ it, but the LSP spec certainly allows it.